server/routes/api/documents.ts

```diff
@@ -795,82 +801,132 @@ router.post("documents.search_titles", auth(), pagination(), async (ctx) => {
   };
 });
 
-router.post("documents.search", auth(), pagination(), async (ctx) => {
-  const {
-    query,
-    includeArchived,
-    includeDrafts,
-    collectionId,
-    userId,
-    dateFilter,
-  } = ctx.body;
-  const { offset, limit } = ctx.state.pagination;
-  const { user } = ctx.state;
+router.post(
+  "documents.search",
+  auth({
+    required: false,
+  }),
+  pagination(),
+  async (ctx) => {
+    const {
+      query,
+      includeArchived,
+      includeDrafts,
+      collectionId,
+      userId,
+      dateFilter,
+      shareId,
+      minWords,
+      maxWords,
+    } = ctx.body;
+    assertNotEmpty(query, "query is required");
+
+    const { offset, limit } = ctx.state.pagination;
+    // this typing is a bit ugly, would be better to have the auth middleware
+    // define the data type for state.user
+    const { user }: { user: User | undefined } = ctx.state;
```

Comment:
It's available – type as `ctx: ContextWithState` to get what you need

Reply:
it's not quite right 

it defines state as
```typescript
  state: {
    user: User;
    token: string;
    authType: "app" | "api";
  };
```
User is not optional here, it works well for auth required but not for auth optional. It also wants to become a generic I think, because body is "unknown" and creates a lot of ts errors when trying to access something on it. I think it wants to be something like 

```typescript
  state: {
    user?: User;
    token?: string;
    authType?: "app" | "api";
  };
  body: any;
;
```
but that will blow the scope of this PR up a bit farther than we want.

Reply:
😅  so. much. tech. debt.